### PR TITLE
UI tweaks

### PR DESCRIPTION
**Pull request by Aid19801**
Various UI Tweaks for the Network Explorer

1. BondBreakdown (this code is from the UPUNK to PUNK decimalisation PR which still needs merging)

2. ContentCard - bringing in mode from context to manage Light/Dark mode.

3. CustomColumnHeading - overwrites the native MUI column headings for Datagrid instances.

4. Nav overhaul flow:-
  i) `originalNavOptions` is mapped through and for each object an `<ExpandableButton />` for the nav is rendered.
  ii) The Nav's expandable buttons are icons to the left of the Nav which are _always_ there/visible. So clicking on one requires the `openDrawer` function and `drawIsOpen` boolean (as well as the nav option's object, url/title, and any nested options it should expand to show the user etc).
  iii) The `<ExpandableButton />` wants to know if a) the nav option has further nested buttons (goverened by whether`nested` array exists) and b) whether the link for the nav option isExternal or not (governed by whether the url has `http` in it) and then decides on styling (nested routes are a lighter colour than top level, selected routes go a darker colour and use a branded orange bar on the right of the nav option).
  iv) Clicking any of the icons fires the `openDrawer()` func, and because they sit within the parent `<ListItem component={Link} to={myNewLocation} />` the routing kicks in also.
  v) If you have open nested items, but the drawer shuts from above (the parent/Chevron arrow) the useEffect will hear the `drawIsOpen` change and close the nested expansion.
  vi) Conversely, if you click the icon for a route with nested routes, it will `openDrawer()` and toggle the nested from false to true to surface the nested routes.

5. Universal Datagrid - added booleans to manage the height so it responds to number of rows rather than hardcoded heights. Killed the MUI filter menu.

6. WorldMap - color changes to match theming / designs.

7. main.tsx / Contexts - added loading states  to Mixnode Detail Info and Delegations api responses. This is used for better handling of spinners/error messaging in Detail page specifically Bond Breakdown.

8. Various SVGs housed in React components - done to have control of color/theme/light/dark mode.
9. Index.html - added Google Font open sans; Arguably we should download and bundle this as ttf. Open to discussion.
10. index.tsx - colors added to palette, general color/theme work.
11. PageGateways - formatting of Datagrid to match Mixnodes (and therefore Figma designs)
12. TwoColTable - added light/dark mode theme
13. PageMixnodes - formatting font sizes etc to match Figma
14. PageOverview - Adding SVGs to Mixnode, Gateway, Validator cards.
15. styles.css - doesnt appear to be an easy way to nuke the buttons on side of Column headers in Datagrid. Have killed them via global css. Again, open to discussion if there's something i've missed.
16. webpack fall back files were necessary as our version of webpack doesn't natively support nodeJs modules like `stream` which appear as dependencies of the Validator pkg.

### CODE REVIEW

File: explorer/package.json [hunk 1]
```diff
@@ -3,12 +3,14 @@
   "version": "1.0.0",
   "private": true,
   "dependencies": {
+    "@cosmjs/math": "^0.26.2",
```
Comment (fmtabbara) [inline explorer/package.json:6]: Can we remove and uninstall this if it's not being used?

File: explorer/src/components/BondBreakdown.tsx [hunk 2]
```diff
@@ -11,103 +11,144 @@ import Paper from '@mui/material/Paper';
...
 
-                {delegations?.data !== undefined && delegations?.data[0] && (
-                    <Table sx={{ minWidth: 650 }} aria-label='delegation totals'>
-                        <TableHead>
+    if (mixnodeDetailInfo?.isLoading) {
+        return <p>Loading...</p>
+    }
+
+    if (showError) {
+        return <p>We are unable to retrieve a Mixnode with that ID. Please try later or Contact Us.</p>
```
Comment (fmtabbara) [inline explorer/src/components/BondBreakdown.tsx]: Perhaps this should be an Alert component from mui?

File: explorer/src/components/BondBreakdown.tsx [hunk 3]
```diff
@@ -11,103 +11,144 @@ import Paper from '@mui/material/Paper';
...
+            const rawTotalBondsAmount = rawDeligationAmount + rawPledgeAmount;
+            return (num * 100 / rawTotalBondsAmount).toFixed(1)
+        }
+    }
 
-                {delegations?.data !== undefined && delegations?.data[0] && (
-                    <Table sx={{ minWidth: 650 }} aria-label='delegation totals'>
-                        <TableHead>
+    if (mixnodeDetailInfo?.isLoading) {
+        return <p>Loading...</p>
```
Comment (fmtabbara) [inline explorer/src/components/BondBreakdown.tsx]: Spinner for consistency?

File: explorer/src/components/BondBreakdown.tsx (hunk 2 above)
Comment (fmtabbara) [inline explorer/src/components/BondBreakdown.tsx]: Perhaps this should be an Alert component from mui?

File: explorer/src/components/BondBreakdown.tsx [hunk 4]
```diff
@@ -11,103 +11,144 @@ import Paper from '@mui/material/Paper';
...
+    }
+
+    if (showError) {
+        return <p>We are unable to retrieve a Mixnode with that ID. Please try later or Contact Us.</p>
+    }
+
+    if (!showError && allContentLoaded) {
+        return (
+            <>
+                <TableContainer component={Paper} sx={{ background: theme => mode === 'dark' ? theme.palette.secondary.dark : theme.palette.primary.light }}>
```
Comment (fmtabbara) [inline explorer/src/components/BondBreakdown.tsx]: I take it this doesnt work without `background: theme => mode === 'dark' ? theme.palette.secondary.dark : theme.palette.primary.light`?

Just curious as I thought using the component={Paper} would have sorted it 

Fine to keep as is.

File: explorer/src/components/Nav.tsx [hunk 5]
```diff
@@ -100,82 +97,135 @@ const Drawer = styled(MuiDrawer, {
...
+  const location = useLocation();
+
+  // colors for nav
+  // TO-DO move to theme if possible (eg divider and orange/brand may
+  // be harder; look into how to add own set of branded colours to palette.
+  const mainNav = '#242C3D';
+  const selectedNotNested = '#111826';
+  const otherNested = '#3C4558';
+  const brandOrange = '#FB6E4E';
+  const fadedDivider = 'rgba(255, 255, 255, 0.1)';
```
Comment (fmtabbara) [inline explorer/src/components/Nav.tsx]: Can these be moved into the theme file and imported? Doesnt matter if you cant add to the theme but lets keep global themeing in one place.

File: explorer/src/components/WorldMap.tsx [hunk 6]
```diff
@@ -31,27 +31,29 @@ export const WorldMap: React.FC<MapProps> = ({ title, countryData, userLocation,
...
             WebkitFilter: hasNoContent ? 'blur(5px)' : null,
             filter: hasNoContent ? 'blur(5px)' : null,
+            width: "100%", height: "auto",
           }}
+          width={800}
+          height={350}
           projectionConfig={{
-            rotate: [-10, 0, 0],
-            scale: 180,
+            // rotate: [-10, 0, 0],
```
Comment (fmtabbara) [inline explorer/src/components/WorldMap.tsx]: Remove comment

File: explorer/src/icons/NymLogoSVG.tsx [hunk 7]
```diff
@@ -9,7 +9,7 @@ export const NymLogoSVG: React.FC = (props) => {
       viewBox="0 0 5389.9 5389.9"
       width="40px"
       height="40px"
-      style={{ margin: theme.spacing(2) }}
+      // style={{ margin: theme.spacing(2) }}
```
Comment (fmtabbara) [inline explorer/src/icons/NymLogoSVG.tsx]: remove comment

**Review by fmtabbara — APPROVED** [review]

File: explorer/package.json (hunk 1 above)
Comment (mmsinclair) [reply explorer/package.json:6]: We are going to have to leave this for now. See https://github.com/nymtech/nym/pull/821#discussion_r731678887.

**Pull request merged**